### PR TITLE
[WIP] Swift Plugin Architecture Proposal

### PR DESCRIPTION
**Pull request by BrunoMazzo**
### Checklist
- [ ] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [ ] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [ ] I've updated the documentation if necessary.

### Motivation and Context

It's a work in progress to enable discussion and feedback. This PR would enable FaslaneSwift link third-parties swift frameworks to run as pluggins. It's better described at issue #11995 

### Description

Basically we change to Marathon to build and run the swift. For more info see issue #11995.

### CODE REVIEW

**Comment by BrunoMazzo** [comment]
I signed it!

**Comment by BrunoMazzo** [comment]
I signed it!

**Comment by BrunoMazzo** [comment]
The author of the commits is me. It's only another account that I you on the same machine and because of this it make this mistake.

**Comment by joshdholtz** [comment]
@BrunoMazzo Will look into why googlebot is struggling here 💪

**Comment by joshdholtz** [comment]
@BrunoMazzo It appears that you may have not signed the CLA with the same email address that your commits were made with (which can be different than the email address your GitHub account is linked to). `git config --global user.email` or `git config user.email` should match the email you signed the CLA with as well 🙃

**Comment by BrunoMazzo** [comment]
I signed it!

**Comment by wade0317** [comment]
maybe you need reset you git config email , then  re- commit , and add pull request agmain.

**Comment by mortenholmgaard** [comment]
Status on this?

**Comment by mortenholmgaard** [comment]
@BrunoMazzo What does this mean - is the feature dropped?
